### PR TITLE
Refactor MOSNS/MOSS algorithms and fix README

### PR DESCRIPTION
**Pull request by MadHypnofrog**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -13,39 +13,36 @@ Install with
...
 
 ::
 
    from sklearn.datasets import load_iris
-   from filters.UnivariateFilter import * # provides you a filter class, basic measures and cutting rules
+   from ITMO_FS.filters import UnivariateFilter, GLOB_MEASURE, GLOB_CR # provides you a filter class, basic measures and cutting rules
 
    data, target = load_iris(True)
-   res = UnivariateFilter("SpearmanCorr", GLOB_CR["Best by value"](0.9999)).run(data, target)
+   res = UnivariateFilter(GLOB_MEASURE["SpearmanCorr"], GLOB_CR["Best by value"](0.9999)).run(data, target)
```
Comment (LastShekel) [inline README.rst]: Better to call Spearman Corr
and select by value

File: ITMO_FS/embedded/MOSS.py [hunk 2]
```diff
@@ -2,15 +2,19 @@
 
 import numpy as np
 from imblearn.over_sampling import SMOTE
-from .filter_utils import augmented_rvalue#, SMOTE
+from sklearn.linear_model import SGDClassifier
+from ITMO_FS.utils import augmented_rvalue
 
 
 class MOSS(object):
```
Comment (LastShekel) [inline ITMO_FS/embedded/MOSS.py]: Why not to make just function like run or something like that

File: ITMO_FS/embedded/MOSS.py [hunk 3]
```diff
@@ -22,19 +26,20 @@ class MOSS(object):
...
 
-    def __init__(self, seed=42):
+    def __init__(self, loss='log', seed=42):
+        if loss not in ['hinge', 'log']:
+            raise KeyError("Loss should be 'hinge' or 'log', %r was passed" % loss)
+        self.loss = loss
         rnd.seed = seed
 
-    def run(self, X, y, alpha=0.5, nu=0.0001, threshold=10e-4, epochs=10000):
+    def run(self, X, y, alpha=0.5, threshold=10e-4, epochs=100000):
```
Comment (LastShekel) [inline ITMO_FS/embedded/MOSS.py]: Why do we need epochs?

Comment (LastShekel) [reply ITMO_FS/embedded/MOSS.py]: And pass model itself

File: ITMO_FS/embedded/MOSNS.py [hunk 4]
```diff
@@ -56,22 +59,13 @@ def run(self, X, y, alpha=0.5, nu=0.0001, threshold=10e-4, epochs=10000):
...
-                # in random data, noise features usually have higher values of b while important features have close-to-zero values
-                oldB = b
-                oldB0 = b0
-                for i in range(X.shape[0]):
-                    if 1 - y[i] * (np.dot(oldB, X[i]) + oldB0) > 0:
-                        b += nu * y[i] * X[i]
-                        b0 += nu * y[i]
-                b -= nu * (l * alpha * np.sign(oldB) + l * (1 - alpha) * oldB)
-            b /= np.max(b)
+        for l in np.arange(0.0002, 0.02, 0.0002):  # TODO: do a little more research on the range of lambdas
```
Comment (LastShekel) [inline ITMO_FS/embedded/MOSNS.py]: pass alpha too

File: ITMO_FS/embedded/MOSS.py (hunk 2 above)
Comment (MadHypnofrog) [reply ITMO_FS/embedded/MOSS.py]: discussed in pm, decided to keep the class for now and merge both algorithms into one class

File: ITMO_FS/embedded/MOSS.py (hunk 3 above)
Comment (MadHypnofrog) [reply ITMO_FS/embedded/MOSS.py]: the number of epochs is passed in the model as max_iter
added the option to pass the model in constructor

File: ITMO_FS/embedded/MOSNS.py (hunk 4 above)
Comment (MadHypnofrog) [reply ITMO_FS/embedded/MOSNS.py]: added

File: README.rst (hunk 1 above)
Comment (MadHypnofrog) [reply README.rst]: changed to function call

File: ITMO_FS/embedded/MOS.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            The range of lambdas that should form the regularization path.
+        Returns
+        ------
+        array-like, shape (n_samples,n_selected_features) : the resulting dataset with remaining features
+        """
+        min_rvalue = 1
+        min_b = []
+        for l in lambdas:  # TODO: do a little more research on the range of lambdas
+            model = self.model(loss=self.loss, random_state=rnd.seed, penalty='elasticnet', 
+                alpha=l, l1_ratio=alpha, max_iter=epochs)
```
Comment (LastShekel) [inline ITMO_FS/embedded/MOS.py]: that's strange naming for alpha parameter

File: ITMO_FS/embedded/MOS.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            The range of lambdas that should form the regularization path.
+        Returns
+        ------
+        array-like, shape (n_samples,n_selected_features) : the resulting dataset with remaining features
+        """
+
+        newX, newY = SMOTE(random_state=rnd.seed).fit_resample(X, y)
+        return self.runMOSNS(newX, newY, alpha, threshold, epochs, lambdas)
+
+    def runMOSNS(self, X, y, alpha=0.5, threshold=10e-4, epochs=100000, lambdas=np.arange(0.0002, 0.02, 0.0002)):
```
Comment (LastShekel) [inline ITMO_FS/embedded/MOS.py]: better to remove epochs because it's parameter of model

File: ITMO_FS/embedded/MOS.py (hunk 5 above)
Comment (MadHypnofrog) [reply ITMO_FS/embedded/MOS.py]: changed according to python naming

File: ITMO_FS/embedded/MOS.py (hunk 6 above)
Comment (MadHypnofrog) [reply ITMO_FS/embedded/MOS.py]: discussed a bit and decided to keep it for now bcs there's no need to use kwargs here, changed the value to 1000 (default value for SGD) though

**Pull request merged**